mautic/mautic
# Add the stage and owner fields to the contact fields list

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:


The `stage`, `owner's email` and `owner's id` fields are added to form field mapping for the `contact` object. Values of the `stage` and `owner` fields of a contact will be changed after form submission.

![image](https://github.com/mautic/mautic/assets/96085911/e780b141-3be2-405c-8c3f-2f0238d7b098)


**Currently WIP, because I'm not 100% sure how it will impact other code. And perhaps there is a better solution for setting the owner and stage as contact fields on a form**


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create new form or edit the existing one 
3. Add a new field with type `Email`
4. Match the form fields with the contact field `owner's email` or `owner's id`
5. Add a new field with type `Text`
6. Match the form field with the contact field `stage`
7. Open the form, fill in the fields and submit
8. Check the contact. 
* The stage is selected by name or id. If no stage is founf by the entered data, the exception will be thrown (it was an existing behaviour)
* The owner is selected by email. If no user is found by the entered email address, no owner will be set, owner field remains the same. Empty or if an existing owner exists then that will be used.




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11248"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
